### DISCUSSION TITLE
Breezy Weather tries to access device sensors whenever you look at the main screen

### DISCUSSION THREAD

**Discussion by matchboxbananasynergy** [breezy-weather/breezy-weather#415] [Q&A] [answered]
Hey there,

I think I might already have an idea of what's happening here, but I wanted to confirm my suspicions.

I use GrapheneOS, which introduces an additional permission for apps, so that you're able to deny access to sensors that are currently not handled by other standard Android permissions.

A description of the permission can be found [here](https://grapheneos.org/features#sensors-permission-toggle):

>Sensors permission toggle: disallow access to all other sensors not covered by existing Android permissions (Camera, Microphone, Body Sensors, Activity Recognition) including an accelerometer, gyroscope, compass, barometer, thermometer and any other sensors present on a given device. When access is disabled, apps receive zeroed data when they check for sensor values and don't receive events. GrapheneOS creates an easy to disable notification when apps try to access sensors blocked by the permission being denied. This makes the feature more usable since users can tell if the app is trying to access this functionality.
>
>To avoid breaking compatibility with Android apps, the added permission is enabled by default. When an app attempts to access sensors and receives zeroed data due to being denied, GrapheneOS creates a notification which can be easily disabled. The Sensors permission can be set to be disabled by default for user installed apps in Settings ➔ Privacy.

I've set it so new apps have the sensors permission disabled by default via the setting mentioned above, so I receive a notification when an app that has that permission denied to it tries to access those sensors, so that I can enable them if needed.

Therefore, I noticed that Breezy Weather consistently tries to access the sensors whenever I look at the main screen.

My *assumption* is that this is due to some animation that probably moves depending on the phone's position, and therefore requires access to the sensors; is that assumption correct?

If so, I wonder whether that should be the default setting. Not because of GrapheneOS users getting that notification (we can make it so it doesn't appear again for that app and just keep sensors denied), because if the reason *is* that it is just for an animation, it feels like the app shouldn't be accessing that for such a novelty purpose if it can be avoided.

If my assumption was wrong, feel free to clue me in; thanks!

**Comment by papjul** [comment, marked as answer]
Animations are a core part of this app so it will be kept as default. You are free to disable sensor in main screen settings, or have a look at more simple weather apps if that's what you are looking for.

**Reply by matchboxbananasynergy** [reply]
Thanks for the reply. That's totally understandable. Just so that's clear to me, are the sensors only being used for the animations, or are there other non-visual features which use them?

If it's purely visual, I can keep the permission denied. :)

**Reply by papjul** [reply]
Main screen and live wallpaper use them for the same feature. I don't remember it being used elsewhere.

I would suggest turning sensor off from settings instead of relying on this non-standard permission as it was not tested that way.
